lab/environments.py

```diff
@@ -391,6 +396,7 @@ def __init__(
         self.qos = qos
         self.time_limit_per_task = time_limit_per_task
         self.memory_per_cpu = memory_per_cpu
+        self.cpus_per_task = cpus_per_task or 1
```

Comment:
I think it's best to use the default already in the constructor (instead of `None`).